presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -830,6 +832,12 @@ else if (isPushedDownSubfield(column)) {
                     columnMappings.add(columnMapping);
                     regularIndex++;
                 }
+                // ROW_ID is synthesized but not prefilled. We don't have a columnmappingkind for this.
+                // Might need to add one.
+                // else if (isRowIdColumnHandle(column)) {
```

Comment:
uncomment this code so you handle rowids

Reply:
let me work more on this. As is, if I uncomment it, the tests fail. I think fully resolving this might require landing https://github.com/prestodb/presto/pull/22251 first.  The current state of this PR is a good checkpoint where everything passes. If there's something actively broken here (and there well might be) then there's a missing test I need to write.